clang/lib/DPCT/MigratePythonSetupScript.cpp

```diff
@@ -0,0 +1,215 @@
+//===-------------------- MigratePythonSetupScript.cpp --------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "MigratePythonSetupScript.h"
+#include "Diagnostics.h"
+#include "Error.h"
+#include "PatternRewriter.h"
+#include "SaveNewFiles.h"
+#include "Statics.h"
+#include "Utility.h"
+
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/raw_os_ostream.h"
+
+#include <algorithm>
+#include <cstring>
+#include <fstream>
+#include <string>
+
+using namespace clang::dpct;
+using namespace llvm::cl;
+
+namespace path = llvm::sys::path;
+namespace fs = llvm::sys::fs;
+
+static std::vector<clang::tooling::UnifiedPath /*file path*/>
+    PythonSetupScriptFilesSet;
+static std::map<clang::tooling::UnifiedPath /*file path*/,
+                std::string /*content*/>
+    PythonSetupScriptFileBufferMap;
+static std::map<clang::tooling::UnifiedPath /*file name*/, bool /*is crlf*/>
+    ScriptFileCRLFMap;
+
+static std::map<std::string /*Python setup syntax*/,
+                MetaRuleObject::PatternRewriter /*Python setup migraiton rule*/>
+    PythonSetupBuildInRules;
+
+static std::map<std::string /*file path*/,
+                std::vector<std::string> /*warning msg*/>
+    FileWarningsMap;
+
+void collectPythonSetupScripts(const clang::tooling::UnifiedPath &InRoot,
```

Comment:
The main logic is the same as https://github.com/tomflinda/SYCLomatic/blob/c1595f5689d74c1d7a4b14b37d91451b510d37f4/clang/lib/DPCT/MigrateCmakeScript.cpp#L118,  suggest to extract a common function to MigrateBuildScript.cpp/h file.

Reply:
Done